### PR TITLE
pkg/manifests: etcd defaults to disabled

### PR DESCRIPTION
**Pull request by squat**
cc @brancz

### CODE REVIEW

File: pkg/manifests/manifests_test.go [hunk 1]
```diff
@@ -723,7 +723,7 @@ func TestPrometheusEtcdRulesFiltered(t *testing.T) {
 }
 
 func TestPrometheusEtcdRules(t *testing.T) {
-	c, err := NewConfigFromString(`etcd: {}`)
+	c, err := NewConfigFromString(`etcd: {enabled: true}`)
```
Comment (brancz) [inline pkg/manifests/manifests_test.go]: Are these changes actually necessary? The empty object is also non-nil, I think we actually dropped the `enabled` field.

Comment (squat) [reply pkg/manifests/manifests_test.go]: The enables field still exists: https://github.com/squat/cluster-monitoring-operator/blob/1bb2db07a7aac0773d14457d4a74d880914db990/pkg/manifests/config.go#L104. We can re-adjust the semantics to always enable etcd Monitoring if the object is non-nil, but I wanted to make the pr small

Comment (brancz) [reply pkg/manifests/manifests_test.go]: Sorry, I was under the impression we did remove it. Let's keep it as is and fix the bug at hand.

**Comment by squat** [comment]
@brancz I just rebased this after it went stale due to some recent changes. PTAL :)

**Comment by brancz** [comment]
thanks for taking care of this! :+1: 

/lgtm
